go/mcap/writer.go

```diff
@@ -833,20 +837,26 @@ func NewWriter(w io.Writer, opts *WriterOptions) (*Writer, error) {
 	compressed := bytes.Buffer{}
 	var compressedWriter *countingCRCWriter
 	if opts.Chunked {
-		switch opts.Compression {
-		case CompressionZSTD:
+		switch {
+		case opts.Compressor != nil: // must be top
+			if opts.Compression == "" {
```

Comment:
My first reaction is that the interface feels a little strange with 'compressor' and 'compression' decoupled. I don't see where `Compression` is used in the case of a custom compressor, but presumably an lz4 writer should always be accompanied by type LZ4. Is `Compression` required if a compressor is passed? If so, would it make sense to have `Compressor` contain the compression, and embed the ResettableWriteCloser type?

Reply:
it's an option, but what I don't really like about it is it forces all consumers of this to implement their own Compressor type (to expose a `Compression() string` method), whereas the current interface has a chance of matching the compressors out of the box - although I think this will only be true of one of lz4 and zstd, depending on whether or not Reset() is chosen to send an error out.

Reply:
but we can't make that change here unfortunately